projects/vanilla/src/app/configurators/metadata-configurator.component.ts

```diff
@@ -7,43 +7,132 @@ import { ComponentConfig, ConfiguratorContext } from "@sinequa/ngx-ui-builder";
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="filterable" [(ngModel)]="metadataToModify.filterable" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="filterable">Filterable</label>
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="excludable" [(ngModel)]="metadataToModify.excludable" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="excludable">Excludable</label>
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="collapseRows" [(ngModel)]="metadataToModify.collapseRows" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="collapseRows">Collapse rows</label>
+      </div>
+      <div class="form-check mb-2" *ngIf="isEntity">
+        <input class="form-check-input" type="checkbox" id="showEntityExtract" [(ngModel)]="metadataToModify.showEntityExtract" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="showEntityExtract">Show entity extract</label>
+      </div>
+      <div class="mb-2" *ngIf="isEntity">
+        <label class="form-label" for="entityExtractMaxLines">Max lines for entity extract</label>
+        <input type="number" class="form-control" id="entityExtractMaxLines" [(ngModel)]="metadataToModify.entityExtractMaxLines">
+      </div>
+    </ng-container>
   </div>
 </div>
     `
 })
-export class MetadataConfiguratorComponent implements OnChanges {
+export class MetadataConfiguratorComponent {
   @Input() context: ConfiguratorContext;
-  @Input() metadata: MetadataConfig[];
+  @Input() metadata: string[];
+  @Input() metadataConfig: MetadataConfig[];
 
+  modifiedMetadata: string;
+  metadataString: string;
+  metadataToModify: MetadataConfig;
   isEntity: boolean;
 
-  constructor(public appService: AppService) {}
-
-  ngOnChanges(): void {
-    this.isEntity = this.appService.isEntity(this.config.item);
-  }
+  constructor(public appService: AppService) { }
 
   get config(): ComponentConfig {
     return this.context.config;
   }
 
-  configChanged(isItem?: boolean) {
-    if (isItem) {
-      this.isEntity = this.appService.isEntity(this.config.item);
-    }
+  get metadataSelection(): string[] {
+    const stringItems = this.config.metadataConfig?.filter(c => !c.field) || [];
+    return this.metadata.concat(stringItems);
+  }
+
+  get updatableMetadata(): MetadataConfig[] {
+    return this.config.metadataConfig?.filter(c => !!c.field) || [];
+  }
+
+  insertString() {
+      if (this.metadataString.trim() !== "") {
+        this.config.metadataConfig.push(this.metadataString);
+        this.config.items.push(this.metadataString);
+        this.metadataString = '';
+        this.configChanged();
+      }
+  }
+
+  configChanged() {
     this.context.configChanged();
+    setTimeout(() => {
```

Comment:
Hmm is there a good reason for this setTimeout?

Reply:
It's actually for what's inside, since we trigger `configChanged`, the previous object for `this.metadataToModify` becomes readonly, and so I need to fetch the new related object, and if I don't put a `setTimeout` it still gets a readonly value, and not with it 🤷

Reply:
What do you mean by "readonly"? Maybe "metadataToModify" should a string ?

Reply:
I see what you're doing here, but the danger is that the settimeout gets executed outside of Angular's lifecycle, which can lead to other issues. Maybe a better way: if the config update generates a new `context` object, then `ngOnChanges` will be called, so that could one place to handle it?

Reply:
That did the trick! 👍